### PR TITLE
Use Spark provided entrypoint

### PR DESCRIPTION
**Pull request by akhurana001**
Changes:

- Adds support for spark entrypoint via the spark-distribution: https://github.com/apache/spark/blob/branch-2.4/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh 
- Also deletes  broken flytekit_spark_entrypoint.sh in favor of the above.

## Type
 - [x] Bug Fix

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/409

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by kumare3** [comment]
you will have to increment the version in flytekit/__init__.py

File: scripts/flytekit_spark_entrypoint.sh [hunk 1]
```diff
@@ -88,30 +88,30 @@ case "$SPARK_K8S_CMD" in
...
       --deploy-mode client
-      "$@" "$PYSPARK_PRIMARY" "$PYSPARK_ARGS"
+      "$@" $PYSPARK_PRIMARY $PYSPARK_ARGS
     )
     ;;
     driver-r)
     CMD=(
       "$SPARK_HOME/bin/spark-submit"
       --conf "spark.driver.bindAddress=$SPARK_DRIVER_BIND_ADDRESS"
       --deploy-mode client
```
Comment (kumare3) [inline scripts/flytekit_spark_entrypoint.sh:98]: Lets put comments over here explaining what we are doing. Else this is hard to follow

File: scripts/flytekit_spark_entrypoint.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ myuid=$(id -u)
 mygid=$(id -g)
 # turn off -e for getent because it will return error code in anonymous uid case
 set +e
-uidentry=$(getent passwd "$myuid")
+uidentry=$(getent passwd $myuid)
```
Comment (kumare3) [inline scripts/flytekit_spark_entrypoint.sh:27]: comments throughout

Comment (akhurana001) [reply scripts/flytekit_spark_entrypoint.sh:30]: I am tagging this as deprecated in favor of official spark entrypoint in the spark distribution. We can also remove this if no-one is using it (given its broken in its current form).

**Comment by kumare3** [comment]
👍 LGTM
